runtime/jcl/common/java_lang_Class.cpp

```diff
@@ -1881,29 +1947,43 @@ Java_java_lang_Class_getNestMembersImpl(JNIEnv *env, jobject recv)
 		J9ClassLoader *classLoader = clazz->classLoader;
 
 		for (i = 0; i < nestMemberCount; i++) {
-			J9Class *nestMember = NULL;
 			J9UTF8 *nestMemberName = NNSRP_GET(nestMembers[i], J9UTF8 *);
 
 			PUSH_OBJECT_IN_SPECIAL_FRAME(currentThread, resultObject);
 			nestMember = vmFuncs->internalFindClassUTF8(currentThread, J9UTF8_DATA(nestMemberName), J9UTF8_LENGTH(nestMemberName), classLoader, J9_FINDCLASS_FLAG_THROW_ON_FAIL);
 			resultObject = POP_OBJECT_IN_SPECIAL_FRAME(currentThread);
 
 			if (NULL == nestMember) {
+				/* If internalFindClassUTF8 fails to find the nest member, it sets
+				 * a NoClassDefFoundError
+				 */
 				goto _done;
-			} else if (nestMember->nestHost != nestHost) {
-				setNestmatesIncompatibleClassChangeError(currentThread, nestMember, nestHost);
+			} else if (NULL == nestMember->nestHost) {
+				nestLoadStatus = vmFuncs->loadAndVerifyNestHost(currentThread, nestMember, 0);
+				if (J9_VISIBILITY_ALLOWED != nestLoadStatus) {
+					goto _done;
+				}
+			}
+			if (nestMember->nestHost != nestHost) {
+				nestLoadStatus = J9_VISIBILITY_NEST_MEMBER_NOT_CLAIMED_ERROR;
 				goto _done;
-			} else {
-				J9JAVAARRAYOFOBJECT_STORE(currentThread, resultObject, i + 1, J9VM_J9CLASS_TO_HEAPCLASS(nestMember));
 			}
+			J9JAVAARRAYOFOBJECT_STORE(currentThread, resultObject, i + 1, J9VM_J9CLASS_TO_HEAPCLASS(nestMember));
 		}
 	}
 
 	result = vmFuncs->j9jni_createLocalRef(env, resultObject);
 
 _done:
+	if (J9_VISIBILITY_ALLOWED != nestLoadStatus) {
```

Comment:
This code should be moved into `loadAndVerifyNestHost()`.  Callers who allow `loadAndVerifyNestHost` to throw should expect the ICCE to be pending already in the failure case and shouldn't have to set it themselves.

This makes the contract of `loadAndVerifyNestHost` simpler - an exception will be pending in all failure cases if the caller allows exceptions to be thrown - and simplifies the callers as they don't have to duplicate the exception call everywhere.